MapboxVision/src/main/java/com/mapbox/vision/manager/DelegateVisionManager.kt

```diff
@@ -23,43 +28,84 @@ internal interface DelegateVisionManager : BaseVisionManager {
     val isStarted: Boolean
     val isCreated: Boolean
 
+    var visionEventsListener: VisionEventsListener?
+    var keepListenerStrongRef: Boolean
+    var videoSource: T
+
     fun create(
         nativeVisionManagerBase: NativeVisionManagerBase,
         performanceManager: PerformanceManager
     )
 
-    fun start(
-        visionEventsListener: VisionEventsListener,
-        onCountrySet: (Country) -> Unit = {}
-    )
+    fun start(onCountrySet: (Country) -> Unit = {})
 
     fun stop()
     fun destroy()
 
     fun setVideoSourceListener(videoSourceListener: VideoSourceListener)
 
+    fun attachVideoSourceListener(videoSourceListener: VideoSourceListener)
+    fun detachVideoSourceListener()
+
     fun setModelPerformanceConfig(modelPerformanceConfig: ModelPerformanceConfig)
 
+    fun getFrameStatistics(): FrameStatistics
+
+    fun checkManagerCreated()
+    fun checkManagerStarted()
+
+    fun addVideoSourceListener(listener: VideoSourceListener)
+    fun removeVideoSourceListener(listener: VideoSourceListener)
+
     fun worldToPixel(worldCoordinate: WorldCoordinate): PixelCoordinate?
     fun pixelToWorld(pixelCoordinate: PixelCoordinate): WorldCoordinate?
     fun worldToGeo(worldCoordinate: WorldCoordinate): GeoCoordinate?
     fun geoToWorld(geoCoordinate: GeoCoordinate): WorldCoordinate?
 
-    fun getFrameStatistics(): FrameStatistics
-
-    fun checkManagerCreated()
-    fun checkManagerStarted()
+    class Impl<T : VideoSource> : DelegateVisionManager<T> {
 
-    class Impl : DelegateVisionManager {
+        override lateinit var videoSource: T
 
         override var externalVideoSourceListener: VideoSourceListener? = null
+            set(value) {
+                value?.let { compositeListenerVideoSource.addListener(it) }
```

Comment:
```suggestion
                value?.let { compositeVideoSourceListener.addListener(it) }
```
we already have _VideoSourceListener_ and imho we shouldn't add any words in the middle, better to use prefix/postfix